py/picca/fitter2/data.py

```diff
@@ -0,0 +1,213 @@
+import fitsio
+from functools import partial
+import numpy as np
+from numpy import linalg
+from . import utils
+from . import pk
+from . import xi
+from scipy.sparse import csr_matrix
+
+class data:
+    def __init__(self,dic_init):
+
+        self.name = dic_init['data']['name']
+        self.tracer1 = dic_init['data']['tracer1']
+        if 'tracer2' in dic_init['data']:
+            self.tracer2 = dic_init['data']['tracer2']
+        else:
+            self.tracer2 = self.tracer1
+
+        self.ell_max = dic_init['data']['ell-max']
+        self.zref = dic_init['data']['zref']
+
+        fdata = dic_init['data']['filename']
+        h = fitsio.FITS(fdata)
+        da = h[1]['DA'][:]
+        co = h[1]['CO'][:]
+        dm = csr_matrix(h[1]['DM'][:])
+        rp = h[1]['RP'][:]
+        rt = h[1]['RT'][:]
+        z = h[1]['Z'][:]
+
+        h.close()
+
+        r = np.sqrt(rp**2+rt**2)
+        mu = rp/r
+
+        rp_min = dic_init['cuts']['rp-min']
+        rp_max = dic_init['cuts']['rp-max']
+
+        rt_min = dic_init['cuts']['rt-min']
+        rt_max = dic_init['cuts']['rt-max']
+
+        r_min = dic_init['cuts']['r-min']
+        r_max = dic_init['cuts']['r-max']
+
+        mu_min = dic_init['cuts']['mu-min']
+        mu_max = dic_init['cuts']['mu-max']
+
+        ## select data within cuts
+        mask = (rp > rp_min) & (rp < rp_max)
+        mask = mask & (rt > rt_min) & (rt < rt_max)
```

Comment:
`mask &= (rt > rt_min) & (rt < rt_max)` is more beautiful

Reply:
beauty!! essential